app/assets/javascripts/discourse/app/pre-initializers/theme-errors-handler.js

```diff
@@ -1,22 +1,43 @@
 import { isTesting } from "discourse-common/config/environment";
-import { getAndClearThemeErrors } from "discourse/app";
+import { getAndClearUnhandledThemeErrors } from "discourse/app";
 import PreloadStore from "discourse/lib/preload-store";
 import getURL from "discourse-common/lib/get-url";
 import I18n from "I18n";
+import { bind } from "discourse-common/utils/decorators";
+
+const showingErrors = new Set();
 
 export default {
   name: "theme-errors-handler",
   after: "inject-discourse-objects",
 
   initialize(container) {
-    const currentUser = container.lookup("current-user:main");
     if (isTesting()) {
       return;
     }
-    renderErrorNotices(currentUser);
-    document.addEventListener("discourse-theme-error", () =>
-      renderErrorNotices(currentUser)
-    );
+
+    this.currentUser = container.lookup("current-user:main");
+
+    getAndClearUnhandledThemeErrors().forEach((e) => {
+      reportThemeError(this.currentUser, e);
+    });
+
+    document.addEventListener("discourse-error", this.handleDiscourseError);
+  },
+
+  cleanup() {
+    document.removeEventListener(this.handleDiscourseError);
+    delete this.currentUser;
+  },
```

Comment:
@davidtaylorhq I just reviewed our pre-intializers and have a question about this function, since I didn't catch that earlier. Is it called at all? Should that be a `teardown()` instead?

Reply:
🤦 right, will fix